src/poisson/simulation.jl

```diff
@@ -3,7 +3,7 @@ function Base.rand(
 ) where {M,T<:Real}
     mark_dist = mark_distribution(pp)
     N = rand(rng, Poisson(float(ground_intensity(pp) * (tmax - tmin))))
-    times = rand(rng, Uniform(tmin, tmax), N)
+    times = sort(rand(rng, Uniform(tmin, tmax), N))
```

Comment:
Good catch!
```suggestion
    times = sort!(rand(rng, Uniform(tmin, tmax), N))
```
This would be even more efficient because it sorts in-place

Reply:
I agree !